### PR TITLE
Bookmarks fixes

### PR DESCRIPTION
**Pull request by henpe071**
Bookmarks can be organized easier. They now appear where they are drop instead of at the bottom of the list.
Click outside a bookmark to deselect.
Fix delete so bookmarks are remove from the project when deleted.
Bookmarks save the state of the video when the bookmark was created and double clicking a bookmark will jump to that point in the video with the saved state.

Continuation of #55 

TODO 
Remove images from the folder when bookmarks are removed.
Add context menu in the categories.

### CODE REVIEW

File: ViAn/GUI/Bookmark/bookmarklist.cpp [hunk 1]
```diff
@@ -19,9 +19,21 @@ BookmarkList::BookmarkList(bool accept_container, int container_type, QWidget* p
...
-
     setIconSize(QSize(ImageGenerator::THUMBNAIL_SIZE, ImageGenerator::THUMBNAIL_SIZE));
-    connect(this, &BookmarkList::itemDoubleClicked, this, &BookmarkList::on_double_clicked);
+
+    clear();
+}
+
+BookmarkList::~BookmarkList() {
+    for (int i = 0; i < count(); ++i) {
+        if (item(i)->type() == BOOKMARK) {
```
Comment (fredrikiselius) [inline ViAn/GUI/Bookmark/bookmarklist.cpp]: I think this if statement could be reduced by simply checking for either BOOKMARK or CONTAINER and then simply deleting item(i) without casting?

`if (item(i)->type() == BOOKMARK || item(i)->type() == CONTAINER)`
    `delete item(i);`

File: ViAn/GUI/Bookmark/bookmarklist.cpp [hunk 2]
```diff
@@ -132,15 +183,25 @@ void BookmarkList::rename_item(){
...
-    emit set_bookmark_video(b_item->get_bookmark()->get_video_project(), b_item->get_frame_number());
+    if (currentItem()->type() == BOOKMARK) {
+        BookmarkItem* bm_item = dynamic_cast<BookmarkItem*>(currentItem());
+        Bookmark* b_mark = bm_item->get_bookmark();
+        b_mark->get_video_project()->remove_bookmark(b_mark);
+        delete bm_item;
+    } else if (currentItem()->type() == CONTAINER) {
+        BookmarkCategory* cat_item = dynamic_cast<BookmarkCategory*>(currentItem());
+        // Remove all bookmarks from the disputed list
+        for (auto bm_item : cat_item->get_disputed()) {
```
Comment (fredrikiselius) [inline ViAn/GUI/Bookmark/bookmarklist.cpp:197]: Shouldn't `bm_item` be deleted here as well?

File: ViAn/GUI/Bookmark/bookmarkwidget.cpp [hunk 3]
```diff
@@ -25,25 +20,31 @@ BookmarkWidget::BookmarkWidget(QWidget *parent) : QWidget(parent){
...
 //    bm_list->setContextMenuPolicy(Qt::CustomContextMenu);
 //    connect(bm_list, SIGNAL(customContextMenuRequested(QPoint)), this, SLOT(item_context_menu(QPoint)));
 }
 
-void BookmarkWidget::add_new_folder() {
-    BookmarkCategory* f2 = new BookmarkCategory(std::string("Category " +  std::to_string(category_cnt++)), bm_list, CONTAINER);
-    bm_list->addItem(f2);
-    connect(f2, SIGNAL(set_bookmark_video(VideoProject*,int)), this, SIGNAL(play_bookmark_video(VideoProject*,int)));
 
+// TODO Move to List?
```
Comment (fredrikiselius) [inline ViAn/GUI/Bookmark/bookmarkwidget.cpp]: Makes more sense there

File: ViAn/Project/videoproject.cpp [hunk 4]
```diff
@@ -236,10 +226,30 @@ void VideoProject::remove_analysis(BasicAnalysis *analysis) {
...
+    m_unsaved_changes = true;
+    bookmark->remove_exported_image();
+    delete bookmark;
+}
+
 /**
  * @brief VideoProject::delete_artifacts
  * Delete bookmark files.
- * TODO "UNUSED"
+ * TODO "UNUSED" only used in tests
```
Comment (fredrikiselius) [inline ViAn/Project/videoproject.cpp:252]: Then it should be removed

File: ViAn/Project/videoproject.cpp [hunk 5]
```diff
@@ -253,13 +263,13 @@ void VideoProject::delete_artifacts(){
     m_unsaved_changes = true;
 }
 
-// TODO, not used
+// TODO, only used in tests
```
Comment (fredrikiselius) [inline ViAn/Project/videoproject.cpp:266]: This one as well

File: ViAn/GUI/Bookmark/bookmarklist.cpp (hunk 2 above)
Comment (henpe071) [reply ViAn/GUI/Bookmark/bookmarklist.cpp:197]: It is deleted. This functions deletes the category which in the desctructor deletes the lists in the category (disp & ref) and then in the destructor for bookmarklist (with your first comment) the items are deleted.

File: ViAn/Project/videoproject.cpp (hunk 4 above)
Comment (henpe071) [reply ViAn/Project/videoproject.cpp:252]: I put these as an issue an make a seperate pull request for that.

**Pull request merged**